src/web/components/elements/InteractiveBlockComponent.tsx

```diff
@@ -57,18 +63,171 @@ and is sent with all interactive elements in the scriptUrl from CAPI to do a two
 It has not been updated since 2016.
 
 MIGRATION FROM FRONTEND
-- TODO For the standard boot.js, we will re-write the behavior in modern JS to avoid the requirement of an AMD loader
+- For the standard boot.js, we have re-written the behavior in modern JS to avoid the requirement of an AMD loader
 and to avoid loading the boot.js file.
 - For all other files that do not load the standard boot.js, we'll add a AMD loader to the page
 
 For the remaining few we dynamically load and AMD loader and support the contract as defined with curl AMD loader
 
 */
+const supportingMinHeight = 200;
+const inlineMinHeight = 500;
+const getMinHeight = (role: string, loaded: boolean) => {
+	if (loaded) {
+		return `auto`;
+	}
+	return role === 'supporting'
+		? `${supportingMinHeight}px`
+		: `${inlineMinHeight}px`;
+};
+const wrapperStyle = (role: string, loaded: boolean) => css`
+	${body.medium()};
+	background-color: ${neutral[100]};
+	font-weight: 300;
+	min-height: ${getMinHeight(role, loaded)};
+	position: relative;
+`;
+
+const placeholderLinkStyle = css`
+	position: absolute;
+	bottom: 1rem;
+	left: 1rem;
+`;
+
+// https://interactive.guim.co.uk/embed/iframe-wrapper/0.1/boot.js
+const setupWindowListeners = (iframe: HTMLIFrameElement) => {
+	// Calls func on trailing edge of the wait period
+
+	const postMessage = (message: Record<string, unknown>) => {
+		if (iframe.contentWindow) {
+			iframe.contentWindow.postMessage(JSON.stringify(message), '*');
+		}
+	};
 
-export const InteractiveBlockComponent = ({ url, scriptUrl, alt }: Props) => {
+	window.addEventListener(
+		'message',
+		(event) => {
+			if (event.source !== iframe.contentWindow) {
+				return;
+			}
+
+			const message: Record<string, unknown> = JSON.parse(event.data);
```

Comment:
Maybe set a try catch around that `parse`? I'm not sure what control we have oover `event.data` but if it were to contain a function or just not be valid in general...